### PR TITLE
Contribution docs

### PR DESCRIPTION
**Pull request by stephanieatte**
This PR adds some key sections and documents that have been omitted. It allows users to run tests with docker and make test without docker compose

- Add the Contribution Section back to the repo
- Remove the docker-compose.yml file back to the repo
- Add Updated Dockerfile to the root folder
- Updated Makefile

### CODE REVIEW

**Review by jeremybumsted — COMMENTED** [review]
Testing this locally, `make test` from the contribution docs and `docker compose run --rm plugin_test` seem to fail with bats issues.

```shell
bats_load_safe: Could not find '/usr/local/lib/bats/load.bash'[.bash]
```

 @stephanieatte Not sure if anything needs to be updated there or if it's a me issue? Have you run the tests locally to make sure everything works?

File: docker-compose.yml
Comment (pzeballos) [inline docker-compose.yml:12]: This was removed because it's unused and it's referencing monebag

Comment (stephanieatte) [reply docker-compose.yml:12]: Yes @pzeballos

**Comment by stephanieatte** [comment]
Will check on the contribution docs to make sure it works just like it should

**Comment by stephanieatte** [comment]
@jeremybumsted please can you try to test this again

**Comment by stephanieatte** [comment]
@jeremybumsted @pzeballos i have made the changes please can you confirm that it works on your end

File: Dockerfile
Comment (stephanieatte) [reply Dockerfile:8]: Thanks @pzeballos will replace it with goreleaser release --skip=validate  as seen in https://goreleaser.com/deprecations#-skip

Comment (pzeballos) [reply Dockerfile:8]: Oh yeah, that one too 🙂

**Review by jeremybumsted — CHANGES_REQUESTED** [review]
@stephanieatte awesome stuff

I think we're still needing to change some things in the actual [CONTRIBUTING](https://github.com/buildkite-plugins/monorepo-diff-buildkite-plugin/blob/master/CONTRIBUTING.md#developing) doc as we reference GitHub actions when it's now all done via the plugin pipeline, and the entire [testing](https://github.com/buildkite-plugins/monorepo-diff-buildkite-plugin/blob/master/CONTRIBUTING.md#testing) section will probably need an update.

File: CONTRIBUTING.md
Comment (pzeballos) [inline CONTRIBUTING.md:19]: This is not true ("will automatically run ..."). We have the block step that a maintainer will need to unblock to run the tests

File: Dockerfile
Comment (pzeballos) [inline Dockerfile:1]: This should remain in the `tests\Dockerfile` because it doesn't build an image to run the project but instead runs the tests. 
Or even have this in a pipeline

File: CONTRIBUTING.md
Comment (pzeballos) [inline CONTRIBUTING.md:20]: I would remove this, because we can start reviewing the PR even if the tests are failing. We should help the contributor

**Review by pzeballos — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md
Comment (jeremybumsted) [inline CONTRIBUTING.md:25]: Do we do anything with these coverage reports? It may be good to have those added to the build as an artifact/annotation so that we can also check them and make sure we have that coverage.

**Review by jeremybumsted — APPROVED** [review]

File: CONTRIBUTING.md
Comment (stephanieatte) [reply CONTRIBUTING.md:25]: For now we don't export it anywhere just to check if its up to the threshold. Will look into either options

**Comment by stephanieatte** [comment]
Make all the necessary changes

- Moved the dockerfile to tests/
- updated Contributions docs

@jeremybumsted

**Review by jeremybumsted — CHANGES_REQUESTED** [review]
Slight wording suggestion

**Review by pzeballos — DISMISSED** [review]
The default filename for Dockerfile it's with the uppercase D, can you change `tests/dockerfile`  to `tests/Dockerfile?`

I also agree with Jeremy's suggestions

**Comment by stephanieatte** [comment]
@pzeballos @jeremybumsted all changes made

**Review by jeremybumsted — APPROVED** [review]
@stephanieatte it looks good to me - only thing I would suggest is changes based on my earlier comment: https://github.com/buildkite-plugins/monorepo-diff-buildkite-plugin/pull/36/files#r1868508453

**Review by pzeballos — APPROVED** [review]
👏🏻

**Pull request merged**
